core_impl/src/main/java/io/opencensus/implcore/stats/MeasureToViewMap.java

```diff
@@ -92,47 +93,52 @@ synchronized void registerView(View view, Clock clock) {
   }
 
   // Records stats with a set of tags.
-  synchronized void record(TagContext tags, MeasureMap stats) {
+  synchronized void record(TagContext tags, MeasureMap stats, Timestamp timestamp) {
```

Comment:
Why does record() always need a timestamp?

Reply:
IntervalView needs timestamps to determine which bucket the stats should be put into, thus we need to pass in the timestamp for each method that calls IntervalView.record(). We could have another record() method that doesn't need timestamp for CumulativeView, but since this is an implementation detail, I just chose to have one generic record() for both Interval and Cumulative views.

Reply:
As discussed, for all of our current measures, we always record both cumulative and interval stats so the timestamp needs to be created so there is no overhead. Eventually, we may want to optimize cumulative stats but this should be fine for now.